integration-tests/tests/radx-catalog-test.js

```diff
@@ -0,0 +1,33 @@
+const _ = require('lodash/fp')
+const { registerTest } = require('../utils/jest-utils')
+const { withUserToken } = require('../utils/terra-sa-utils')
+const { enableDataCatalog, navigateToLibrary } = require('../utils/integration-helpers')
+
+
+const testRadXCatalogFn = withUserToken(async ({ page, testUrl, token }) => {
+  let testSuccess = true
+  await navigateToLibrary(page, testUrl, token)
+  await enableRadX(page)
+  try {
+    await enableDataCatalog(page)
+    testSuccess = false
```

Comment:
@nawatts I am not super thrilled about this approach to testing, do you have a good approach for this sort of test failure?

Reply:
You could use:
```js
await expect(() => enableDataCatalog(page)).rejects.toThrow()
```

https://jestjs.io/docs/expect#rejects

Also, it would be preferable to test this with a unit test instead of an integration test to reduce the overhead of running the test. This test relies on the query for the data catalog checkbox to time out and (on my machine) takes ~40 seconds to run.

Reply:
Is there going to be any sort of intro to the unit test framework we use? I have been having trouble understanding how to handle unit tests.

Reply:
@msilva-broad is working on some ([UIE-6](https://broadworkbench.atlassian.net/browse/UIE-6) / [8](https://broadworkbench.atlassian.net/browse/UIE-8)).

In the meantime, there's the docs for the various tools:
- [Jest](https://jestjs.io/)
- [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/)

Here's how this can be tested with a unit test (change Datasets.js to export the Datasets component and put this code in Datasets.test.js):
```js
import '@testing-library/jest-dom'

import { render } from '@testing-library/react'
import { h } from 'react-hyperscript-helpers'
import { isRadX } from 'src/libs/brand-utils'
import * as Nav from 'src/libs/nav'
import { Datasets } from 'src/pages/library/Datasets'


jest.mock('src/libs/brand-utils', () => ({
  ...jest.requireActual('src/libs/brand-utils'),
  isRadX: jest.fn()
}))

jest.mock('src/libs/nav', () => ({
  ...jest.requireActual('src/libs/nav'),
  getLink: jest.fn(),
  useRoute: jest.fn()
}))

beforeEach(() => {
  Nav.useRoute.mockReturnValue({ title: 'Datasets', params: {}, query: {} })
})

describe('Datasets', () => {
  it('shows a toggle for the data catalog', () => {
    const { getByLabelText, getByText } = render(h(Datasets))
    expect(getByText('Preview the new Data Catalog')).toBeTruthy()
    expect(getByLabelText('New Catalog OFF')).toBeTruthy()
  })

  it('does not show a toggle when viewed as RadX', () => {
    isRadX.mockReturnValue(true)

    const { queryByLabelText, queryByText } = render(h(Datasets))
    expect(queryByText('Preview the new Data Catalog')).toBeFalsy()
    expect(queryByLabelText('New Catalog OFF')).toBeFalsy()
  })
})
```